esmvalcore/preprocessor/_area.py

```diff
@@ -315,6 +315,25 @@ def _select_representative_point(shape, lon, lat):
     return select
 
 
+def _correct_coords_from_shapefile(cube, cmor_coords):
+    """Get correct lat and lon from shapefile."""
+    lon = cube.coord(axis='X').points
+    lat = cube.coord(axis='Y').points
+    if cube.coord(axis='X').ndim < 2:
+        lon, lat = np.meshgrid(lon, lat, copy=False)
+
+    if not cmor_coords:
+        # Wrap around longitude coordinate to match data
+        lon_180 = np.where(lon >= 180., lon - 360., lon)
```

Comment:
```suggestion
        lon[lon >= 180.] -= 360.
```
would be a bit more concise